src/MergeContext.js

```diff
@@ -954,12 +962,50 @@ class PullRequest {
         return false;
     }
 
+    async _getMergeCommit() {
+        const mergeSha = await GH.getReference("pull/" + this._prNumber() + "/merge");
+        return await GH.getCommit(mergeSha);
+    }
+
+    // obtains abandoned staging checks
+    async _getAbandonedStagingStatuses() {
+        assert(!this._stagedSha());
+
+        const allEvents = await GH.getIssueEvents(this._prNumber());
+        // staging events are PR events where the bot user created a commit referencing this PR
+        let stagingEvents = allEvents.filter(ev => ev.event === "referenced" && ev.actor.login === Config.githubUserLogin());
+        if (!stagingEvents.length)
+            return;
+
+        // just in case: events should be already in chronological order already
+        stagingEvents = stagingEvents.sort((ev1, ev2) => Date.parse(ev1.created_at) - Date.parse(ev2.created_at));
+        const lastStagingEvent = stagingEvents[stagingEvents.length - 1];
+        const mergeCommit = await this._getMergeCommit();
+        const mergeCommitCreatedAt = Date.parse(mergeCommit.author.date); // for merge commits author.date and committer.date should be the same
```

Comment:
> is there any reason why they should different

Just my personal (and possibly wrong) interpretation of what "merge commit" is. To me, that commit brings PR changes (authored _some time ago_) into the target repository branch. Thus, I would expect the author date to reflect something from the PR past rather than current time. However, I would not be surprised if somebody who knows a lot more about the mechanics of these merge commits would explain why the dates must be the same. It does not seem natural/intuitive to me, but sometimes the best approach is not natural or intuitive, especially from an outsider point of view.

To avoid misunderstanding, the problem here is _not_ what GitHub (currently) does.

* One problem is how we describe what we think it (currently) does. We describe it as if it is some self-evident truth or obvious principle, but we seem to be unable to substantiate that.
* Another problem is that given a choice between two dates, if we need to know when the merge commit was _created_ at, the committer date seems to be a more natural choice (when we look at commits where the dates differ, it is the committer date that we would pick). This is especially true if we would prefer to use the updated/new date if the merge commit was modified for some reason (but still represents a true merge commit). I know we do not expect such modifications, but let's assume, for the sake of the argument, that they happened to a merge commit -- which date would we pick then?

Reply:
> the mechanics of these merge commits

I don't think that there is some special magic (or hidden mechanics) in this merge commits - except the fact that they are automatically generated by GitHub. For example, I see nothing special in the commit data (fetched via an API call) except the fact that it has two parent commits. I we create a new (ordinary) branch commit - we expect that author.date=committer.date, right? We create a new commit from scratch - as an atomic operation and both dates ought to be identical. Merge commit is created by GitHub, but the same reasoning should be applicable there.

Reply:
When I create an ordinary commit, the dates are naturally the same because no good/reliable information is available about the changes I authored. Git could look at file timestamps instead of using current time, I guess. That would be natural too, but probably not worth the hassle in the default/common use case.

When I create a merge commit, then my expectations would depend on the type of a merge:

* If I am creating a true merge commit (which is what GitHub appears to be doing based on your "two parents" observations), then, yes, the dates would naturally be the same because the commit object represents the decision to merge (the changes have their own metadata in one of the parent chains).
* I did not know that GitHub merge commit is always a true merge commit. For example, I recalled banning merge commits in GitHub configuration some time ago and expected that to result in squashed merge commits (single parent). With squash merges, the commit author date should not be that of the commit creation date IMO (but GitHub may disagree, of course).

Reply:
> Eduard: If we create a new (ordinary) branch commit - we expect that author.date=committer.date, right?

FWIW, when I cherry-pick a commit, the dates differ:

```
commit d68eb83 (bag51)
AuthorDate: Tue Nov 28 09:34:46 2023 -0500
CommitDate: Thu Nov 30 16:36:15 2023 -0500

    Bug 5274: Successful tunnels logged as TCP_TUNNEL/500
...    
    Cherry-picked SQUID-932-bug5274-no-retryOrBail-for-committed-srv-closure
    commit e12d6d6
```

To me, a merge commit is more similar to a cherry-picked commit than to an ordinary branch commit.